### ISSUE TITLE
JetStream: allow replicas > current cluster size as long as there is quorum

### ISSUE THREAD

**Issue by kozlovic** [nats-io/nats-server#2239]
## Feature Request
Allow replicas number greater than current cluster size as long as there is quorum

#### Use Case:
Say there is a cluster of 3 nodes, if one creates a stream with a replicas of 3, it works.
Then one server is shutdown, there is still quorum and this stream can be used.
However, trying to create another stream with replicas(3) would fail with "insufficient resources".
As shown above a stream with R(3) still can function as long as there is quorum, so it would be nice to be able to create the stream and use it even if there is currently not enough servers to match the replicas count, but there is still quorum.

#### Proposed Change:
Allow creation of assets with a replicas > current cluster size as long that there is quorum.

#### Who Benefits From The Change(s)?
MQTT feature inside the NATS Server creates a stream for every new session (or clean session). If it is determined that the cluster is of size 3 for instance, the assets are created with this replicas number. Which means that it does not work if a server is not running. Reducing the replicas count to quorum size is an issue too, for instance in this case we would create with R(2), but then if one of the server where the asset is stored goes down, then MQTT is unusable.

#### Alternative Approaches
When creating assets, if we get "insufficient resources", we can reduce the replicas count, but then that brings other issues that now this asset may be unusable if the other server is restarted but one of the server hosting the R(2) asset goes down.

**Comment by ripienaar** [comment]
This sounds like a bug to me rather than freature request.  Running with 2 servers in a previously 3 node cluster should not impact anything.

**Comment by kozlovic** [comment]
What I mean is that if one tries to *create* a stream with R3 when only 2 servers are running (but there is quorum because it was originally a 3 nodes cluster). Are you saying that in that situation it should still be working? I would like it to work this way for sure, which that is why I describe that it makes a difference in behavior just based on when the stream is created.

**Comment by ripienaar** [comment]
Yeah, I am 100% saying that should be working today and if it does not, thats a bug not a new feature.

**Comment by derekcollison** [comment]
We should be a bit careful here. I generally agree we can do better, but I am also concerned around scenarios like a 3 node cluster and being asked to create a R=5 stream. 

What I think might work is that if we have seen in the past that the cluster size was N, then anything up to N (sans limit of 5) is ok. But if you ask to create a stream with R=5 on a properly constructed cluster of just three nodes that should fail IMO.
